File: izmk-core/src/main/java/ovo/xsvf/izmk/event/annotations/EventPriority.java
package ovo.xsvf.izmk.event.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * @author LangYa466
 * @since 2025/2/16
 */
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface EventPriority {
    int value() default 10;
}

File: izmk-core/src/main/kotlin/ovo/xsvf/izmk/event/annotations/EventPriority.kt
package ovo.xsvf.izmk.event.annotations

/**
 * @author LangYa466
 * @since 2025/2/16
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class EventPriority(val value: Int = 10)